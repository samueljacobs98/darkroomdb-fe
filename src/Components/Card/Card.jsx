import { Link } from "react-router-dom";
import "./Card.scss";

const Card = ({ film }) => {
  return (
    <Link to={{ pathname: `/film/${film.filmId}` }}>
      <div className="card">{film.name}</div>
    </Link>
  );
};

export default Card;
