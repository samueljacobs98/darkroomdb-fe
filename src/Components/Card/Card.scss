@use "../../styles/variables" as *;

.card {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 12.5rem;
  height: 15.625rem;
  padding: 1.25rem;
  text-align: center;
  background-color: rgba($colour-primary, 0.2);
  border: 1px solid rgba($colour-tertiary, 0.5);

  &:hover {
    cursor: pointer;
  }
}
