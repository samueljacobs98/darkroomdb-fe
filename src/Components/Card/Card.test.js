import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Card from "./Card";

it("should match component snapshot", () => {
  const { container } = render(
    <MemoryRouter>
      <Card film={{ filmId: 1, name: "name" }} />;
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
