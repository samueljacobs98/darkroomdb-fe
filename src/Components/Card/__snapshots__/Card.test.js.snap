// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match component snapshot 1`] = `
<div>
  <a
    href="/film/1"
  >
    <div
      class="card"
    >
      name
    </div>
  </a>
  ;
</div>
`;
